components/renku_data_services/notifications/core.py

```diff
@@ -19,3 +23,63 @@ def validate_alert_patch(patch: apispec.AlertPatch) -> models.AlertPatch:
     return models.AlertPatch(
         resolved=patch.resolved,
     )
+
+
+def transform_alert_to_unsaved_alert(alert: apispec.AlertmanagerAlert) -> models.UnsavedAlert | None:
+    """Transform a single Alertmanager alert to an UnsavedAlert."""
+    labels = alert.labels
+    annotations = alert.annotations
+
+    user_id = labels.get("safe_username")
+    if not user_id:
+        logger.warning("Alert is missing 'safe_username' label, skipping: %s", alert)
+        return None
+
+    session_name = labels.get("statefulset")
+    if not session_name:
+        session_name = None
+
+    title = annotations.get("title")
+    if not title:
+        logger.warning("Alert is missing 'title' annotation, skipping: %s", alert)
+        return None
+
+    message = annotations.get("description")
+    if not message:
+        logger.warning("Alert is missing 'description' annotation, skipping: %s", alert)
+        return None
+
+    event_type = labels.get("alertname")
+    if not event_type:
+        logger.warning("Alert is missing 'alertname' label, skipping: %s", alert)
+        return None
+
+    return models.UnsavedAlert(
+        title=title,
+        message=message,
+        event_type=event_type,
+        user_id=user_id,
+        session_name=session_name,
+    )
+
+
+def transform_alertmanager_webhook(
```

Comment:
The function name is not really intuitive to grasp what it is going to do.

maybe `webhook_to_alerts` or `transform_webhook_to_alerts`